src/cryptography/hazmat/primitives/ciphers/aead.py

```diff
@@ -22,21 +25,33 @@ def __init__(self, key: bytes):
             )
         utils._check_byteslike("key", key)
 
-        if len(key) != 32:
+        if len(key) != self._KEY_LEN:
             raise ValueError("ChaCha20Poly1305 key must be 32 bytes.")
 
+        self._cipher_name = aead._aead_cipher_name(self)
         self._key = key
+        self._decrypt_ctx = None
+        self._encrypt_ctx = None
 
     @classmethod
     def generate_key(cls) -> bytes:
-        return os.urandom(32)
+        return os.urandom(ChaCha20Poly1305._KEY_LEN)
 
     def encrypt(
         self,
         nonce: bytes,
         data: bytes,
         associated_data: typing.Optional[bytes],
     ) -> bytes:
+        if not self._encrypt_ctx:
+            self._encrypt_ctx = aead._aead_setup_with_fixed_nonce_len(
+                backend,
+                self._cipher_name,
+                self._key,
+                self._NONCE_LEN,
+                aead._ENCRYPT,
+            )
```

Comment:
This is not safe in the face of concurrent users of it. This doesn't even require multi-threading for it to occur. All it woul take is:


```py
e1 = ctx.encryptor()
e2 = ctx.encryptor()
e1.update(data1)
e2.update(data2)
```

Reply:
Thanks for the quick feedback. The specific use case I've mentioned in the PR is from `asyncio` and the `nonce` is being incremented each time so there isn't a concurrency concern.

I'm not very familiar with this code base. Hopefully you can help me out with suggestion on how to proceed? Thanks!